python-package/xgboost/data.py

```diff
@@ -161,12 +162,12 @@ def _maybe_np_slice(data: np.ndarray, dtype) -> np.ndarray:
 
 
 def _from_numpy_array(
-    data,
-    missing,
-    nthread,
+    data: DataType,
+    missing: float,
```

Comment:
float is actually not accurate, we can also take np.float32 and related types

Reply:
How about using `FloatCompatible = Union[float, np.float32, np.float64]` (do we also need to add `np.float16`?)